spec/models/provider_spec.rb

```diff
@@ -588,6 +588,47 @@
     end
   end
 
+  describe "scopes" do
+    describe ".with_findable_courses" do
+      let(:findable_course) do
+        create(:course, site_statuses: [build(:site_status, :findable)])
+      end
+
+      let(:findable_course_with_accrediting_provider) do
+        create(:course, :with_accrediting_provider, site_statuses: [build(:site_status, :findable)])
+      end
+
+      let(:non_findable_course) do
+        create(:course, site_statuses: [build(:site_status)])
+      end
+
+      let(:non_findable_course_with_accrediting_provider) do
+        create(:course, :with_accrediting_provider, site_statuses: [build(:site_status)])
+      end
+
+      before do
+        findable_course
+        findable_course_with_accrediting_provider
+        non_findable_course
+        non_findable_course_with_accrediting_provider
+      end
+
+      subject {
+        described_class.with_findable_courses
+      }
+
+      it "should return only findable courses' provider and/or accrediting provider" do
```

Comment:
You could split this into contexts and it would be easier to read. 

e.g.
``` 
context "when the provider is the accredited body for a course"  do
  it "is returned"
end

context "when the course is delivered by the provider" do
  it "is returned"
end

context "when the course is unrelated to the provider" do
  it "is not returned"
end
```